File: moving_object.py
class MovingObject:
    def __init__(self,x,y,type,map):
        self.x,self.y,self.type,self.map=x,y,type,map
    def move(self,dx,dy,map_width,map_height):
        self.x=(self.x+dx)%map_width
        self.y+=dy

File: simulator/objects/MovingObject.java
package simulator.objects;

public class MovingObject extends GeneralObject {
    
    // Constructor
    public MovingObject(String type,float x,float y,boolean isAlive){
        super(type,x,y,isAlive);
    }

    // Setter
    public void move(float dx,float dy){
        this.x+=dx;
        this.y+=dy;
    }
}
